packages/coderskit/src/atoms/Avatar/Avatar.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { HTMLAttributes } from 'react';
+import styled from '@emotion/styled';
+import classnames from 'classnames';
+
+export type AvatarVariant = 'photo' | 'text';
+export type AvatarShape = 'circle' | 'square';
+
+export interface AvatarProps extends HTMLAttributes<any> {
+  variant?: AvatarVariant;
+  size?: number;
+  shape?: AvatarShape;
+  image?: string;
+  children?: React.ReactNode;
+}
+
+const AvatarBase = styled.div<AvatarProps>(props => {
+  const { shadows } = props.theme;
+  const { size, shape } = props;
+
+  const calcRadius = () => {
+    if (shape === 'circle') {
+      return '100%';
+    } else if (size === 1 || size === 2) {
+      return '4px';
+    } else {
+      // handles size === 3, size === 4 and any other size
+      return '8px';
+    }
+  };
+
+  const calcDimensions = () => {
+    if (size === 1) {
+      return '16px';
+    } else if (size === 2) {
+      return '24px';
+    } else if (size === 3) {
+      return '32px';
+    } else {
+      // handles size === 4 and other size
+      return '40px';
+    }
+  };
+
+  return {
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    outline: 'none',
+    border: 'none',
+    borderRadius: calcRadius(),
+    width: calcDimensions(),
```

Comment:
Skoro `calcDimensions` w obu przypadkach daje ten sam rezultat to warto wykonać ją przed `return` i przypisać jej rezultat do jakiejś zmiennej (np. `dimension`). Tak to funkcja niepotrzebnie wykonuje się 2 razy

Reply:
Poprawione